src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -29113,6 +29113,134 @@ static void M(C c, string? p1, string? p2)
             CompileAndVerify(comp2);
         }
 
+        [Fact, WorkItem(48489, "https://github.com/dotnet/roslyn/issues/48489")]
+        public void NotNullIfNotNull_Return_BinaryOperator()
+        {
+            var lib_cs =
+@"using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [return: NotNullIfNotNull(""x""), NotNullIfNotNull(""y"")] public static C? operator +(C? x, C? y) => null;
+    public static C? operator *(C? x, C? y) => null;
+    [return: NotNullIfNotNull(""x"")] public static C? operator -(C? x, C? y) => null;
+    [return: NotNullIfNotNull(""y"")] public static C? operator /(C? x, C? y) => null;
+}";
+            var source = @"
+class D
+{
+    static void M(C c1, C c2, C? cn1, C? cn2)
+    {
+        (c1 + c2).ToString(); // no warn
+        (c1 + cn2).ToString(); // no warn
+        (cn1 + c2).ToString(); // no warn
+        (cn1 + cn2).ToString(); // warn
+
+        (c1 * c2).ToString(); // warn
+        (c1 * cn2).ToString(); // warn
+        (cn1 * c2).ToString(); // warn
+        (cn1 * cn2).ToString(); // warn
+
+        (c1 - c2).ToString(); // no warn
+        (c1 - cn2).ToString(); // no warn
+        (cn1 - c2).ToString(); // warn
+        (cn1 - cn2).ToString(); // warn
+        
+        (c1 / c2).ToString(); // no warn
+        (c1 / cn2).ToString(); // warn
+        (cn1 / c2).ToString(); // no warn
+        (cn1 / cn2).ToString(); // warn
```

Comment:
Could you please add a test that verifies expected behavior around `+=`/`-=`/`*=`/`/=` with these operators? Both the nullable state of the expression and the nullable state of the LHS of the assignment.

Reply:
Test added. Let me know, if you still find some missing coverage.